clients/src/main/java/org/apache/kafka/clients/producer/internals/TransactionManager.java

```diff
@@ -384,14 +436,19 @@ synchronized boolean isAborting() {
     }
 
     synchronized void transitionToAbortableError(RuntimeException exception) {
```

Comment:
Do we ever use this version of the method?

Reply:
I guess I will need to manually check if the usages are external 😅

Reply:
It's only used in `TransactionManagerTest`. I didn't want to expose the `InvalidStateTransitionHandler` quirkiness the outside world (even the tests).

I went back and forth on this decision more than a couple of times. I'd certainly welcome your input.

Reply:
I think the only concern is that someone uses this method accidentally -- ideally if we add a new place in the code where we transition state, we don't want to assume it is using the external approach. (I think Philip has a PR that might incorrectly use this methond.)

Reply:
The curse of the default arugment :)

Reply:
I've removed the overloaded methods with default arguments in a few places. It's a little noisier now, but hopefully more explicitly clear.